vpr/src/draw/intra_logic_block.cpp

```diff
@@ -237,11 +237,30 @@ static void draw_internal_load_coords(int type_descrip_index, t_pb_graph_node* p
             /* Find the number of instances for each child pb_type. */
             int num_pb = mode.pb_type_children[j].num_pb;
 
+            int num_blocks = num_pb * num_children;
+
+            // determine an optimal number of columns
+            int num_columns = 1;
+            for (int k = 1; k * k <= num_blocks; ++k) {
+                if (num_blocks % k == 0) {
+                    num_columns = k;
+                }
+            }
+            int num_rows = num_blocks / num_columns;
+
+            const int MAX_WIDTH_HEIGHT_RATIO = 2;
+            if (parent_width > parent_height * MAX_WIDTH_HEIGHT_RATIO) {
+                std::swap(num_columns, num_rows);
+            }
+
             for (int k = 0; k < num_pb; ++k) {
+
+                int num_block = j * num_pb + k;
```

Comment:
I think this is a block index.  In VTR we usually use num_<something> to mean the maximum number of something (e.g. a loop limit or vector size).  If this is a block index, I would use blk_num or iblk or blk_idx instead of num_block.